cgs_vmc/wavefunctions.py

```diff
@@ -1199,6 +1255,33 @@ def build_wavefunction(
     rbm_2 = WAVEFUNCTION_TYPES[wavefunction_type].from_hparams(hparams, 'rbm_2')
     return rbm_1 - rbm_2
 
+  if hparams.wavefunction_type == 'sum':
```

Comment:
Here you can pass modified copies of the hparams to the original from_hparams method. i.e.

```
wf_type_a, wf_type_b = hparams.composite_wavefunction_types
activation_a, activation_b = hparams.composite_output_activations
wf_a_hparams = copy.copy(hparams)
wf_b_hparams = copy.copy(hparams)
wf_a_hparams.set_hparam('output_activation', activation_a)
wf_a_hparams.set_hparam('wavefunction_type', wf_type_a)
wf_b_hparams.set_hparam('output_activation', activation_n)
wf_b_hparams.set_hparam('wavefunction_type', wf_type_n)
wf_a = WAVEFUNCTION_TYPES[wf_a_hparams]
wf_b = WAVEFUNCTION_TYPES[wf_b_hparams]
return wf_a + wf_b
```

Reply:
How about something like this:
// Ideally we would use an enum to store these (sum, diff, prod) strings and call them by enum, but since we have only 3 I think it's ok to go with string.

```
if hparams.wavefunction_type in ('sum', 'diff', 'prod'):
  wf_type_a, wf_type_b = hparams.composite_wavefunction_types
  activation_a, activation_b = hparams.composite_output_activations
  wf_a_hparams = copy.copy(hparams)
  wf_b_hparams = copy.copy(hparams)
  wf_a_hparams.set_hparam('output_activation', activation_a)
  wf_a_hparams.set_hparam('wavefunction_type', wf_type_a)
  wf_b_hparams.set_hparam('output_activation', activation_n)
  wf_b_hparams.set_hparam('wavefunction_type', wf_type_n)
  wf_a = WAVEFUNCTION_TYPES[wf_a_hparams]
  wf_b = WAVEFUNCTION_TYPES[wf_b_hparams]
  if hparams.wavefunction_type == 'sum':
    return wf_a + wf_b
  if hparams.wavefunction_type == 'diff':
    return wf_a - wf_b
  if hparams.wavefunction_type == 'prod':
    return wf_a * wf_b
```